.NET/Microsoft.Recognizers.Text.DateTime/Parsers/BaseDateTimeParser.cs

```diff
@@ -251,6 +251,21 @@ private DateTimeResolutionResult MergeDateAndTime(string text, DateObject refere
 
             ret.FutureValue = DateObject.MinValue.SafeCreateFromValue(futureDate.Year, futureDate.Month, futureDate.Day, hour, min, sec);
             ret.PastValue = DateObject.MinValue.SafeCreateFromValue(pastDate.Year, pastDate.Month, pastDate.Day, hour, min, sec);
+
+            // Handle case like "Wed Oct 26 15:50:06 2016" which year and month separated by time.
+            var matchYear = this.config.YearRegex.Match(text.Substring(er2[0].Start + er2[0].Length ?? 0));
```

Comment:
Please declare a var to receive text.Substring(er2[0].Start + er2[0].Length ?? 0) and give a name that describes what this represents.